RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -12,31 +14,104 @@ namespace Rubberduck.Inspections.QuickFixes
     /// </summary>
     public class PassParameterByReferenceQuickFix : QuickFixBase
     {
-        public PassParameterByReferenceQuickFix(ParserRuleContext context, QualifiedSelection selection)
-            : base(context, selection, InspectionsUI.PassParameterByReferenceQuickFix)
+        private Declaration _target;
+
+        public PassParameterByReferenceQuickFix(Declaration target, QualifiedSelection selection)
+            : base(target.Context, selection, InspectionsUI.PassParameterByReferenceQuickFix)
         {
+            _target = target;
         }
 
         public override void Fix()
         {
-            var parameter = Context.GetText();
+            var argCtxt = GetArgContextForIdentifier(Context, _target.IdentifierName);
 
-            var parts = parameter.Split(new char[]{' '},2);
-            if (1 != parts.GetUpperBound(0))
-            {
-                return;
-            }
-            parts[0] = parts[0].Replace(Tokens.ByVal, Tokens.ByRef);
-            var newContent = parts[0] + " " + parts[1];
+            var terminalNodeImpl = GetByValNodeForArgCtx(argCtxt);
 
-            var selection = Selection.Selection;
+            var replacementLine = GenerateByRefReplacementLine(terminalNodeImpl);
 
-            var module = Selection.QualifiedName.Component.CodeModule;
+            ReplaceModuleLine(terminalNodeImpl.Symbol.Line, replacementLine);
+
+        }
+        private ArgContext GetArgContextForIdentifier(ParserRuleContext context, string identifier)
+        {
+            var procStmtCtx = (ParserRuleContext)context.Parent.Parent;
```

Comment:
Without the cast, procStmtCtx is a identified as a RuleContext which does not appear support a 'children' accesor used in the next statement.